lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +45,50 @@ def call
 
         private
 
+        def extract_credentials!
+          if @sshkey_path && @token_path
+            raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+          end
+
+          if @sshkey_path
+            if @sshkey_path == '-'
+              sshkey = $stdin.read
+              keypath = write_key_to_tmpfile(sshkey).path
+              { sshkey_file: keypath }
```

Comment:
nit: is there a reason to switch from `@sshkey_path` to `sshkey_file`? I would just assume it should be called `sshkey_path` for consistency, but there could be reasons I'm not thinking of? I guess `sshkey_path.path` is a little hard on the 👀 ...

Reply:
Sure I can change it. I didn't have a strong reason.